app/src/androidTest/java/com/sdp/movemeet/view/map/MainMapFragmentTest.java

```diff
@@ -63,41 +86,135 @@
 @RunWith(AndroidJUnit4.class)
 public class MainMapFragmentTest {
 
-    private UiDevice uiDevice;
+    private UiDevice uiDevice = UiDevice.getInstance(getInstrumentation());
     private FirebaseAuth fAuth = FirebaseAuth.getInstance();
+    private String TAG = "Main Map Fragment Test";
 
     @Rule
     public GrantPermissionRule mRuntimePermissionRule = GrantPermissionRule.grant(Manifest.permission.ACCESS_FINE_LOCATION);
 
-   @Rule
-   public FragmentTestRule<?, MainMapFragment> fragmentTestRule =
-           FragmentTestRule.create(MainMapFragment.class);
-
+    @Rule
+    public FragmentTestRule<?, MainMapFragment> fragmentTestRule =
+            FragmentTestRule.create(MainMapFragment.class);
 
     @Before
     public void setUp() {
         fAuth = FirebaseAuth.getInstance();
         fAuth.signInWithEmailAndPassword("test@test.com", "password");
-        uiDevice = UiDevice.getInstance(getInstrumentation());
+        Intents.init();
     }
 
     @Test
-    public void mainMapFragmentIsDisplayedAndGMapsNotNull() throws InterruptedException {
+    public void mainMapFragmentIsDisplayedAndMapIsReadyWhenNotNull() throws InterruptedException {
+        MainMapFragment mapFragment = fragmentTestRule.getFragment();
+        mapFragment.getActivity().runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                mapFragment.onMapReady(mapFragment.googleMap);
+            }
+        });
         onView(withId(R.id.fragment_map)).check(matches((isDisplayed())));
     }
 
     @Test
-    public void mainMapFragment_MarkerOnMapForUser() throws UiObjectNotFoundException, InterruptedException {
+    public void markerOnMapForUser() throws UiObjectNotFoundException, InterruptedException {
         waitFor(5000);
         UiObject marker = uiDevice.findObject(new UiSelector().descriptionContains("I am here !"));
         assertNotNull(marker);
     }
 
+    @Test
+    public void displayingUserMarkerSetsPositionMarker() {
+        MainMapFragment mapFragment = fragmentTestRule.getFragment();
+        mapFragment.currentLocation = new Location("current location");
+        mapFragment.currentLocation.setLatitude(0);
+        mapFragment.currentLocation.setLongitude(0);
+
+        mapFragment.getActivity().runOnUiThread(new Runnable() {
```

Comment:
Why do you use runOnUiThread ?

Reply:
The callbacks methods in the map can only run on the main UI thread, so they can't be called directly in the tests. I have to make those calls run on the UI thread, and this is done by using this method !